packages/docs-site/src/library/pages/forms.md

```diff
@@ -0,0 +1,126 @@
+
+```
+import { TextInput } from '@royalnavy/react-component-library
+
+const MyForm = () => {
+  const [colour, setColour] = useState('Orange')
+  const [errorMessage, setErrorMessage] = useState()
+
+  onFieldChange = (event) => {
+    setColor(event.target.value)
+    if (event.target.value.length < 3) {
+      setErrorMessage('Too short)
+    } else {
+      setErrorMessage(null)
+    }
+  }
+
+  return (
+    <form>
+      <TextInput 
+        className={error ? 'is-invalid' : null}
+        errorMessage={errorMessage}
+        label="Colour"
+        name="colour"
+        value={color} 
+        onChange=(onFieldChange) 
+      />
+    </form>
+  )
+}
+```
+
+## Form libraries
+
+Building forms can result in large amounts of boilerplate/repetitive code and forms can grow to become complex once things such as validation are introduced. With this growth in complexity and size, there is the added risk of different developers adopting different patterns or code becoming bloated and difficult to maintain. With this in mind, a small number of frameworks have been created to help the developer and provide patterns to solve common challenges. The two most popular libraries are [Formik](https://jaredpalmer.com/formik/) and [Redux Forms](https://redux-form.com).
+
+### Formik
+
+Formik promises a toolkit to build forms 'without the tears', though it is not always as simple as it promises. Formik is responsible for managing the forms state, triggering events integrating with a validation library.
+
+When using Formik, the initial values for a form are passed into it and the developer specifies the form layout including the fields that should be displayed. For each field the name property must be provided and Formik is responsible for assigning the associated value.
+
+Formik includes a 'Field' component that by default will render a regular textinput and accepts a 'component' property that is responsible for rendering the field. The component library uses this feature to integrate with Formik. It exposes a Formik version of all of the component library form components on an object named 'Formik'.
+
+Below is a version of the earlier form example but uses Formik and the Nelson Standards Toolkit:
+
+```
+import { Formik, Form } from 'formik'
+import { Formik as FormComponents } from '@royalnavy/react-component-library
+import * as Yup from 'yup';
+
+const { TextInput } = FormComponents
+
+const ColorSchema = Yup.object().shape({
+  colour: Yup.string()
+    .min(3, 'Too Short!')
+    .required('Required'),
+});
+
+const MyForm = () => (
+    <Formik 
+      initialValues={{ color: 'orange' }} 
+      onSubmit={(data) => console.log(data)}
+      validationSchema={ColorSchema}
+    >
+      <Form>
+        <TextInput label="Colour" name="colour" />
+      </Form>
+    </form>
+  )
+}
+```
+
+Note: because the name of the object exposed by the component library conflicts with the name of the object exported by Formik, you must alias it to another name.
```

Comment:
Hmm yeah - you are right, this is annoying

Reply:
So do we want to change that or just live with it? No one is using it yet... though it would be a breaking change